src/textual/widget.py

```diff
@@ -92,11 +93,21 @@ def __rich_repr__(self) -> rich.repr.Result:
             yield "name", self.name
         if self.classes:
             yield "classes", self.classes
+        pseudo_classes = self.pseudo_classes
+        if pseudo_classes:
+            yield "pseudo_classes", pseudo_classes
+        yield "outline", self.styles.outline
 
     def __rich__(self) -> RenderableType:
         renderable = self.render_styled()
         return renderable
 
+    def get_pseudo_classes(self) -> Iterable[str]:
+        """Pseudo classes for a widget"""
+        if self._mouse_over:
+            yield "hover"
+        # TODO: focus
```

Comment:
Should this TODO be deleted or fulfilled as part of this PR?

Reply:
Still need to implement a mechanism to handle a focus mechansim, like alt-tab. Will look at the pseudo class then.